features/filesystem/bd/ExhaustibleBlockDevice.cpp

```diff
@@ -79,17 +78,26 @@ int ExhaustibleBlockDevice::erase(bd_addr_t addr, bd_size_t size)
 {
     MBED_ASSERT(is_valid_erase(addr, size));
 
-    // use an erase cycle
-    if (_erase_array[addr / get_erase_size()] > 0) {
-        _erase_array[addr / get_erase_size()] -= 1;
-    }
+    bd_size_t eu_size = get_erase_size();
+    while (size) {
+        // use an erase cycle
+        if (_erase_array[addr / eu_size] > 0) {
+            _erase_array[addr / eu_size] -= 1;
+        }
 
-    if (_erase_array[addr / get_erase_size()] == 0) {
-        // TODO possibly something more destructive here
-        return 0;
+        if (_erase_array[addr / eu_size] == 0) {
+            return 0;
+        }
+
+        int  err = _bd->erase(addr, eu_size);
+        if (err) {
+            return err;
+        }
+        addr += eu_size;
+        size -= eu_size;
```

Comment:
can size become negative and then the while loop wont exit?

Reply:
No, this can't happen. The function checks that the erase is valid (at its start), meaning that the erase size is a multiple of the erase unit size, so the scenario you describe can't happen.

Reply:
I saw the assert now, thanks